File: FrontEnd (C#)/SoftProgBusiness/GestMaterial/BOI/TesisBOImpl.cs
using SoftProgBusiness.GestMaterial.BO;
using SoftProgModel.GestMaterial;
using SoftProgPersistance.GestMaterial.DAO;
using SoftProgPersistance.GestMaterial.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestMaterial.BOI
{
    public class TesisBOImpl : ITesisBO
    {
        private TesisDAO daoTesis;
        public TesisBOImpl()
        {
            daoTesis = new TesisImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoTesis.eliminar(idObjeto);
        }

        public int insertar(Tesis objeto)
        {
            return daoTesis.insertar(objeto);
        }

        public BindingList<Tesis> listarTodos()
        {
            return daoTesis.listarTodos();
        }

        public int modificar(Tesis objeto)
        {
            return daoTesis.modificar(objeto);
        }

        public Tesis obtenerPorId(int idObjeto)
        {
            return daoTesis.obtenerPorId(idObjeto);
        }

        public void validar(Tesis objeto)
        {
            throw new NotImplementedException();
        }
    }
}

File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionMaterial/boImpl/TesisBOImpl.java

package biblioteca.gestionMaterial.boImpl;

import biblioteca.gestionMaterial.bo.TesisBO;
import biblioteca.gestionMaterial.dao.TesisDAO;
import biblioteca.gestionMaterial.mysql.Tesislmpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.gestion_de_material.model.Tesis;

public class TesisBOImpl implements TesisBO{
private final TesisDAO daoTesis;

    public TesisBOImpl() {
        daoTesis = new Tesislmpl();
    }

    @Override
    public int insertar(Tesis objeto) throws Exception {
        return daoTesis.insertar(objeto);
    }

    @Override
    public int modificar(Tesis objeto) throws Exception {
        return daoTesis.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return daoTesis.eliminar(idObjeto);
    }

    @Override
    public Tesis obtenerPorId(int idObjeto) throws Exception {
        return daoTesis.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Tesis> listarTodos() throws Exception {
        return daoTesis.listarTodos();
    }

    @Override
    public void validar(Tesis objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
}
